google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java

```diff
@@ -2447,4 +2489,32 @@ Session getSession(String name) {
     }
     return null;
   }
+
+  static MultiplexedSessionPrecommitToken getResultSetPrecommitToken(ByteString transactionId) {
+    return getPrecommitToken("ResultSetPrecommitToken", transactionId);
+  }
+
+  static MultiplexedSessionPrecommitToken getPartialResultSetPrecommitToken(
+      ByteString transactionId) {
+    return getPrecommitToken("PartialResultSetPrecommitToken", transactionId);
+  }
+
+  static MultiplexedSessionPrecommitToken getExecuteBatchDmlResponsePrecommitToken(
+      ByteString transactionId) {
+    return getPrecommitToken("ExecuteBatchDmlResponsePrecommitToken", transactionId);
+  }
+
+  static MultiplexedSessionPrecommitToken getPrecommitToken(
+      String value, ByteString transactionId) {
+    if (!transactionSequenceNo.containsKey(transactionId)) {
+      transactionSequenceNo.put(transactionId, new AtomicInteger(0));
+    }
+
+    // Generates an incrementing sequence number
+    int seqNum = transactionSequenceNo.get(transactionId).addAndGet(1);
```

Comment:
nit: you can simplify this by calling `transactionSequenceNo.putIfAbsent(..)`

Reply:
another tiny nit: use `AtomicInteger#incrementAndGet()` instead of `AtomicInteger#addAndGet(int)`

Reply:
Thanks. Done